solutions/gold/usaco-1019.mdx

```diff
@@ -0,0 +1,101 @@
+---
+id: usaco-1019
+source: USACO Gold 2020 February
+title: Delegation
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_deleg_gold_feb20.html)
+
+## Explanation
+
+First note that if $K$ is not a divisor of the $N - 1$, then the edges can
+never be evenly partitioned. Thus, we only need to check divisors of $N - 1$.
+
+We can arbitrarily root the tree at node $1$. Let $\mathtt{toBeMerged[i]}$ store the length of the path passes through node $i$ and its length has not reached $K$ yet. Since only $1$ of such path exist, all of the other paths which source from the subtrees of $i$ must be evenly paired.
+
+For each node $j$ such that $j$ is a child of node $i$, there must be another node $k$ such that $\mathtt{toBeMerged[j]+toBeMerged[k]+2} = K$ in order to complete the path. If no such $k$ exist, then we will store the current length of the path in $\mathtt{toBeMerged[i]}$, but there can only be one such $j$.
+
+Since the number of divisors for $N$ is around $\log N$, we will be iterating over all nodes only $\log N$ times. A table for an upper bound of the number of divisors can be found [here](https://wwwhomes.uni-bielefeld.de/achim/highly.txt).
```

Comment:
> Since the number of divisors for $N$ is around $\log N$,

The number of divisors grows much faster than log N. An upper bound is given here: https://math.stackexchange.com/questions/1053062/upper-limit-for-the-divisor-function/1053070#1053070

Reply:
you can say it's $N^{o(1)}$

Reply:
great! thanks